backend/src/filter/validateMatchPropertyCondition.ts

```diff
@@ -0,0 +1,23 @@
+import { ArgumentError } from '../shared/errors';
+import { FilterCondition, MatchPropertyCondition } from '../shared/queries';
+import { checkValidMatchPropertyCondition } from './checkValidMatchPropertyCondition';
+import { formatErrorMessage } from './formatErrorMessage';
+
+export function validateMatchPropertyCondition(
+  condition: FilterCondition
+): MatchPropertyCondition {
+  const result = <MatchPropertyCondition>condition;
```

Comment:
Why is there an instant type change? So why is ne argument type not of type `MatchPropertyCondition`?

Reply:
Thats because the function does in fact a protected cast. The check is in the next line.